lib/sandbox.macos.ml

```diff
@@ -115,14 +123,16 @@ let run ~cancelled ?stdin:stdin ~log (t : t) config result_tmp =
     post_build ~result_dir ~home_dir t >>= fun () ->
     Lwt.return r
   in
+  let () = Lwt.on_failure cancelled (fun _ -> let _ = unmount_fuse t in () )
+  in
   Lwt.on_termination cancelled (fun () ->
     let aux () =
-      if Lwt.is_sleeping proc then (
+      if Lwt.is_sleeping proc then
         match !proc_id with
-          | Some pid -> Macos.kill_all_descendants ~pid
+          | Some pid -> Macos.kill_all_descendants ~pid >>= fun () -> Lwt_unix.sleep 5.0
           | None -> Log.warn (fun f -> f "Failed to find pid…"); Lwt.return ()
-          )
-      else Lwt.return_unit  (* Process has already finished *)
+      else Lwt.return_unit >>= fun () ->
+      post_cancellation ~result_tmp t
```

Comment:
What I meant is that the indentation of `post_cancellation` is confusing: it is actually part of the `else` block here. If you want to execute it regardless of the `Lwt.is_sleeping proc`, you have to enclose the `if … else …` block in a `begin … end` (or parenthesis) block.
I think that's what you want.

Reply:
Yes, I have an updated version with that modification which I haven't pushed yet.